File: library/src/main/java/de/westnordost/osmfeatures/PerCountryFeatureCollection.java
package de.westnordost.osmfeatures;

import java.util.Collection;
import java.util.List;

/** A collection of features grouped by country code */
public interface PerCountryFeatureCollection {

    /** Returns all features with the given country code */
    Collection<Feature> getAll(List<String> countryCodes);

    /** Returns the feature with the given id with the given country code or null if it has not been
     *  found  */
    Feature get(String id, List<String> countryCodes);
}

File: library/src/main/java/de/westnordost/osmfeatures/PerCountryFeatureCollection.kt
package de.westnordost.osmfeatures

/** A collection of features grouped by country code  */
interface PerCountryFeatureCollection {
    /** Returns all features with the given country code  */
    fun getAll(countryCodes: List<String?>): Collection<Feature>

    /** Returns the feature with the given id with the given country code or null if it has not been
     * found   */
    operator fun get(id: String, countryCodes: List<String?>): Feature?
}
